### PR TITLE
Add documentation for the kafka connector

### PR DESCRIPTION
**Pull request by elonazoulay**
Add documentation regarding the integration with
Confluent Schema Registry.

### CODE REVIEW

**Comment by mosabua** [comment]
@kokosing can you please review this first for technical accuracy... I can then follow up with grammar, formatting and such. Thx

File: docs/src/main/sphinx/connector/kafka.rst [hunk 1]
```diff
@@ -269,6 +293,65 @@ Field           Required  Type      Description
...
+========================================================== ======================================================================================================
+
+``kafka.confluent-schema-registry-url``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The comma separated list of schema registry urls used to connect to the schema registry, for example ``http://schema-registry-1.example.org:8081,http://schema-registry-2.example.org:8081``.
+This property is required if the ``CONFLUENT`` table description supplier is used; there is no default.
+
+``kafka.confluent-schema-registry-client-cache-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The size of the local subjects cache.
```
Comment (kokosing) [inline docs/src/main/sphinx/connector/kafka.rst:319]: Can you please elaborate a bit more about what information about subject cache holds?

File: docs/src/main/sphinx/connector/kafka.rst [hunk 2]
```diff
@@ -269,6 +293,65 @@ Field           Required  Type      Description
...
+Confluent Table Description Supplier
+------------------------------------
+The confluent table description supplier uses the `Confluent Schema Registry
+<https://docs.confluent.io/1.0/schema-registry/docs/intro.html>`_. to
+discover table definitions. The benefits of using the schema registry is that new tables can be defined without
+a cluster restart, schema updates will be detected and there is no need to define tables manually.
+
+========================================================== ======================================================================================================
+Property Name                                              Description
+========================================================== ======================================================================================================
```
Comment (kokosing) [inline docs/src/main/sphinx/connector/kafka.rst:305]: Lines here are too long, let's wrap the line somewhere. This go to entire change. Please make sure you wrap about 100.

**Review by kokosing — COMMENTED** [review]
Looks good to me content wise. Some minor comments, also waiting on review of @Praveen2112 and @mosabua.

File: docs/src/main/sphinx/connector/kafka.rst [hunk 3]
```diff
@@ -269,6 +293,65 @@ Field           Required  Type      Description
 
 There is no limit on field descriptions for either key or message.
 
+Confluent Table Description Supplier
```
Comment (kokosing) [inline docs/src/main/sphinx/connector/kafka.rst:296]: Please mention limitations:
 - no INSERT
 - only AVRO

**Comment by mosabua** [comment]
Please ping me after the comments from @kokosing are addressed @elonazoulay . Then I can review in detail. In general please use 80 char hard wrap and prefer list table usage.

**Review by Praveen2112 — APPROVED** [review]

**Comment by mosabua** [comment]
Replaced by https://github.com/trinodb/trino/pull/6791

**Pull request closed without merging**